src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -0,0 +1,498 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import com.google.api.services.genomics.model.Read;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.util.UUID;
+
+/**
+ * Unified read interface for use throughout the GATK.
+ *
+ * Adapter classes implementing this interface exist for both htsjdk's {@link SAMRecord} ({@link SAMRecordToGATKReadAdapter})
+ * and the Google Genomics {@link Read} ({@link GoogleGenomicsReadToGATKReadAdapter})
+ */
+public interface GATKRead extends Locatable {
+
+    /**
+     * @return A globally unique identifier for this read, suitable for use as a key
+     */
+    UUID getUUID();
+
+    /**
+     * @return The name of the read (equivalent to QNAME in SAM), or {@code null} if the read has no name.
+     */
+    String getName();
+
+    /**
+     * Set the name of the read (equivalent to QNAME in SAM), or set to {@code null} if the read has no name.
+     *
+     * @param name new name for the read
+     */
+    void setName( final String name );
+
+    /**
+     * @return The number of bases in the read
+     */
+    int getLength();
+
+    /**
+     * Set the position of the read (the contig and the start position). Cannot be used to
+     * set the read to an unmapped position; use {@link #setIsUnmapped} for that purpose.
+     *
+     * @param contig Contig the read is mapped to
+     * @param start Start position of the read (1-based, inclusive)
+     */
+    void setPosition( final String contig, final int start );
```

Comment:
Null contig not allowed (produces an IllegalArgumentException), since this method cannot be used to set the read to be unmapped. Calling `setPosition()` does explicitly mark the read as mapped.